crates/rpc/rpc/src/eth/api/block.rs

```diff
@@ -2,20 +2,115 @@
 
 use crate::{
     eth::{
-        api::transactions::build_transaction_receipt_with_block_receipts,
+        api::transactions::ReceiptBuilder,
         error::{EthApiError, EthResult},
     },
     EthApi,
 };
+
+use futures::Future;
 use reth_evm::ConfigureEvm;
 use reth_network_api::NetworkInfo;
-use reth_primitives::{BlockId, TransactionMeta};
-use reth_provider::{BlockReaderIdExt, ChainSpecProvider, EvmEnvProvider, StateProviderFactory};
+use reth_primitives::{BlockId, Receipt, SealedBlock, TransactionMeta};
+use reth_provider::{
+    BlockIdReader, BlockReader, BlockReaderIdExt, ChainSpecProvider, EvmEnvProvider,
+    ReceiptProvider, StateProviderFactory,
+};
 use reth_rpc_types::{AnyTransactionReceipt, Header, Index, RichBlock};
 use reth_rpc_types_compat::block::{from_block, uncle_block_from_header};
 use reth_transaction_pool::TransactionPool;
 use std::sync::Arc;
 
+use super::transactions::BuildReceipt;
+
+/// Commonly used block related functions for the [EthApi] type in the `eth_` namespace.
+pub trait EthBlocks {
```

Comment:
I'd like to keep the traits entirely separate, otherwise we could easily lose over

perhaps we put this under a new folder `traits` or similar?